pkg/lib/bundle/generate.go

```diff
@@ -79,69 +80,83 @@ func GenerateFunc(directory, outputDir, packageName, channels, channelDefault st
-	if channels == "" || packageName == "" {
+	if bundleConfig.channels == "" || bundleConfig.packageName == "" {
 		var notProvided []string
-		if channels == "" {
+		if bundleConfig.channels == "" {
 			notProvided = append(notProvided, "channels")
 		}
-		if packageName == "" {
+		if bundleConfig.packageName == "" {
 			notProvided = append(notProvided, "package name")
 		}
 		log.Infof("Bundle %s information not provided, inferring from parent package directory",
 			strings.Join(notProvided, " and "))
 
-		i, err := NewBundleDirInterperter(directory)
+		i, err := NewBundleDirInterperter(bundleConfig.bundleDir)
 		if err != nil {
 			return fmt.Errorf("please manually input channels and packageName, "+
-				"error interpreting bundle from directory %s, %v", directory, err)
+				"error interpreting bundle from directory %s, %v", bundleConfig.bundleDir, err)
 		}
 
-		if channels == "" {
-			channels = strings.Join(i.GetBundleChannels(), ",")
-			if channels == "" {
+		if bundleConfig.channels == "" {
+			bundleConfig.channels = strings.Join(i.GetBundleChannels(), ",")
+			if bundleConfig.channels == "" {
 				return fmt.Errorf("error interpreting channels, please manually input channels instead")
 			}
-			log.Infof("Inferred channels: %s", channels)
+			log.Infof("Inferred channels: %s", bundleConfig.channels)
 		}
 
-		if packageName == "" {
-			packageName = i.GetPackageName()
-			log.Infof("Inferred package name: %s", packageName)
+		if bundleConfig.packageName == "" {
+			bundleConfig.packageName = i.GetPackageName()
+			log.Infof("Inferred package name: %s", bundleConfig.packageName)
 		}
 
-		if channelDefault == "" {
-			channelDefault = i.GetDefaultChannel()
-			if !containsString(strings.Split(channels, ","), channelDefault) {
-				channelDefault = ""
+		if bundleConfig.channelDefault == "" {
+			bundleConfig.channelDefault = i.GetDefaultChannel()
+			if !containsString(strings.Split(bundleConfig.channels, ","), bundleConfig.channelDefault) {
+				bundleConfig.channelDefault = ""
 			}
-			log.Infof("Inferred default channel: %s", channelDefault)
+			log.Infof("Inferred default channel: %s", bundleConfig.channelDefault)
 		}
 	}
 
 	log.Info("Building annotations.yaml")
 
 	// Generate annotations.yaml
-	content, err := GenerateAnnotations(mediaType, ManifestsDir, MetadataDir, packageName, channels, channelDefault)
+	content, err := GenerateAnnotations(mediaType, ManifestsDir, MetadataDir, bundleConfig.packageName,
+		bundleConfig.channels, bundleConfig.channelDefault)
 	if err != nil {
 		return err
 	}
 
 	// Push the output yaml content to the correct directory and conditionally copy the manifest dir
-	outManifestDir, outMetadataDir, err := CopyYamlOutput(content, directory, outputDir, workingDir, overwrite)
+	outManifestDir, outMetadataDir, err := CopyYamlOutput(content, bundleConfig.bundleDir, bundleConfig.outputDir, bundleConfig.overwrite)
 	if err != nil {
 		return err
 	}
 
+	logger := log.NewEntry(log.New())
```

Comment:
Going forward it would be _very_ convenient to configure logging via parameter in exported functions.

Reply:
yes, I would hope so, but I would like to contain this PR scope since this is exactly what is in our validation cmd for now. They should be updated in the same time.